helper/src/main/java/me/lucko/helper/utils/XMaterial.java

```diff
@@ -0,0 +1,1686 @@
+        int versionNumber = Integer.parseInt(StringUtils.replace(version, ".", ""));
+
+        return currentVer >= versionNumber;
+    }
+
+    /**
+     * Gets the exact major version (..., 1.9, 1.10, ..., 1.14)
+     *
+     * @param version Supports {@link Bukkit#getVersion()}, {@link Bukkit#getBukkitVersion()} and normal formats such as "1.14"
+     * @return the exact major version.
+     */
+    @Nonnull
+    public static String getExactMajorVersion(@Nonnull String version) {
+        // getBukkitVersion()
+        if (version.contains("SNAPSHOT") || version.contains("-R"))
+            version = version.substring(0, version.indexOf("-"));
+        // getVersion()
+        if (version.contains("git"))
+            version = StringUtils.replace(version.substring(version.indexOf("MC:") + 4), ")", "");
+        if (version.split(Pattern.quote(".")).length > 2) version = version.substring(0, version.lastIndexOf("."));
+        return version;
+    }
+
+    /**
+     * Parses the string arugment to a version.
+     * Supports {@link Bukkit#getVersion()}, {@link Bukkit#getBukkitVersion()} and normal formats such as "1.14"
+     *
+     * @param version the server version.
+     * @return the Minecraft version represented by the string.
+     */
+    @Nonnull
+    private static MinecraftVersion valueOfVersion(@Nonnull String version) {
+        version = getExactMajorVersion(version);
+        if (version.equals("1.10") || version.equals("1.11") || version.equals("1.12"))
+            return MinecraftVersion.VERSION_1_9;
+        version = StringUtils.replace(version, ".", "_");
+        if (!version.startsWith("VERSION_")) version = "VERSION_" + version;
+        String check = version;
+        return Arrays.stream(MinecraftVersion.VALUES).anyMatch(v -> v.name().equals(check)) ? MinecraftVersion.valueOf(version) : MinecraftVersion.UNKNOWN;
+    }
+
+    /**
+     * Checks if the material can be damaged from {@link #DAMAGEABLE}.
+     *
+     * @param name the name of the material.
+     * @return true of the material can be damaged.
+     */
+    public static boolean isDamageable(@Nonnull String name) {
+        Validate.notEmpty(name, "Material name cannot be null or empty");
+        return Arrays.stream(DAMAGEABLE).anyMatch(name::contains);
+    }
+
+    public static boolean isDuplicated(@Nonnull String name) {
+        Validate.notEmpty(name, "Material name cannot be null or empty");
+        String formatted = format(name);
+        return DUPLICATED.entrySet().stream().anyMatch(x -> x.getKey().name().equals(formatted));
+    }
+
+    /**
+     * Converts the material's name to a string with the first letter uppercase.
+     *
+     * @return a converted string.
+     */
+    @Nonnull
+    public String toWord() {
+        return toWord(this.parseMaterial().name());
+    }
+
+    /**
+     * Gets the ID (Magic value) of the material.
+     * If an {@link IllegalArgumentException} was thrown from this method,
```

Comment:
Additional paragraphs in comments should include an empty line between previous paragraphs.